### PR TITLE
HPCC-32741 Refactor code to work towards a Thor generic disk activity

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith feel free to discuss each change.  They are separate commits to clarify the steps taken, but will be squashed before merging (if you accept them)

File: common/thorhelper/thorcommon.cpp [hunk 1]
```diff
@@ -1074,16 +1074,20 @@ class CRowStreamReader : public CSimpleInterfaceOf<IExtRowStream>
...
-    const byte *getNextPrefetchRow()
+    const byte *getNextPrefetchRow(size32_t & size)
     {
         while (true)
         {
             ++progress;
             if (checkEmptyRow())
+            {
+                size = 0;
                 return nullptr;
```
Comment (jakesmith) [inline common/thorhelper/thorcommon.cpp:1085]: trivial: clearer if break; and fell through to common ending that does same? [as if (checkExitConditions()) does]

File: common/thorhelper/thorcommon.cpp [hunk 2]
```diff
@@ -1248,7 +1255,10 @@ class CRowStreamReader : public CSimpleInterfaceOf<IExtRowStream>
 
     virtual void stop() override
     {
-        stop(NULL);
+        if (!eos) {
```
Comment (jakesmith) [inline common/thorhelper/thorcommon.cpp:1258]: trivial/style: rest of class is not using this k&r style.

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good, 2 trivial comments.

**Comment by ghalliday** [comment]
@jakesmith I have made the two you changes you suggested, squashed, and updatd the commit message.

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good.

**Pull request merged**